sql/analyzer/resolve_tables.go

```diff
@@ -321,16 +306,28 @@ func validateDropTables(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope,
 
 	// validates that each table in DropTable is ResolvedTable and each database of
 	// each table is TableDropper (each table can be of different database later on)
+	var resolvedTables []sql.Node
 	for _, table := range dt.Tables {
-		rt, ok := table.(*plan.ResolvedTable)
-		if !ok {
-			return nil, transform.SameTree, plan.ErrUnresolvedTable.New(rt.String())
-		}
-		_, ok = rt.Database.(sql.TableDropper)
-		if !ok {
-			return nil, transform.SameTree, sql.ErrDropTableNotSupported.New(rt.Database.Name())
+		switch t := table.(type) {
+		case *plan.ResolvedTable:
+			_, ok = t.Database.(sql.TableDropper)
+			if !ok {
+				return nil, transform.SameTree, sql.ErrDropTableNotSupported.New(t.Database.Name())
+			}
+			resolvedTables = append(resolvedTables, table)
+		case *plan.UnresolvedTable:
+			if !dt.IfExists() {
+				return nil, transform.SameTree, sql.ErrUnknownTable.New(t.String())
+			}
```

Comment:
Can we set a warning from the context here? That would match MySQL's behavior (example output below) and also seems nice to leave a clue if we silently drop something (otherwise those can be really tricky to debug if something goes wrong). 

```
mysql> drop table if exists users2;
Query OK, 0 rows affected, 1 warning (0.01 sec)

mysql> show warnings;
+-------+------+-----------------------------+
| Level | Code | Message                     |
+-------+------+-----------------------------+
| Note  | 1051 | Unknown table 'dolt.users2' |
+-------+------+-----------------------------+
1 row in set (0.00 sec)
```